File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int n) {
    long long arr[2];
    arr[0] = 0;
    arr[1] = 1;
    long long temp = 0;
    long long i = 1;
    long long sum = arr[0]+arr[1];
    while(i < n-1)
    {
        temp = arr[1] %1234567;
        arr[1] = (arr[1] %1234567 + arr[0]%1234567)%1234567;
        arr[0] = temp;
        i++;
        sum = (arr[0] +arr[1])%1234567;
        
    }
    return sum;
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    arr = []
    temp = 0
    for i in range(0, n+1):
        if i == 0:
            arr.append(0)
        elif i == 1:
            arr.append(1)
        else:
            temp = arr[1]
            arr[1] = arr[0] + arr[1]
            arr[0] = temp
            
    return arr[1]%1234567